app/jobs/attachments_cleanup_job.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# delete/destroy attachements that have been deleted X days ago
+class AttachmentsCleanupJob < ApplicationJob
+  queue_as :default
+
+  # Finds all deleted attachments more than `days_old`` days old, and destroys them
+  # Params:
+  # +days_old+:: positive integer. Number of days old and older to destroy. Default is 7
+  def perform(days_old = 7)
+    if !days_old.instance_of?(Integer) || (days_old < 1)
+      err = "'#{days_old}' is not a positive integer!"
+      Rails.logger.error err
+      raise err
+    end
+
+    Rails.logger.info "Cleaning up all deleted attachments which are at least #{days_old} days old."
+
+    # SELECT "attachments".* FROM "attachments"
+    #   WHERE "attachments"."deleted_at" IS NOT NULL AND "attachements"."deleted_at" <= $1
+    attachments_to_delete = Attachment.only_deleted.where(deleted_at: ..(Date.yesterday.midnight - days_old.day))
+    attachments_to_delete.each(&:really_destroy!)
```

Comment:
**nitpick (blocking):** Can we update this to call `attachment.file.purge` before calling `attachment.really_destroy!`. This way the file on the storage service is immediately removed instead of being queued to be removed later. By default when `really_destroy!` is called `attachment.file.purge_later` is called.

https://edgeguides.rubyonrails.org/active_storage_overview.html#removing-files

Or we could actually just change the `has_one_attached` to include `.dependent: :purge` as any action by the user will soft delete and permanent removal is only completed via a scheduled task.

https://edgeapi.rubyonrails.org/classes/ActiveStorage/Attached/Model.html#method-i-has_one_attached